docs/topics/testing/tools.txt

```diff
@@ -1651,14 +1651,17 @@ your test suite.
 
     Output in case of error can be customized with the ``msg`` argument.
 
-.. method:: TransactionTestCase.assertQuerysetEqual(qs, values, transform=repr, ordered=True, msg=None)
+.. method:: TransactionTestCase.assertQuerysetEqual(qs, values, transform=None, ordered=True, msg=None)
 
-    Asserts that a queryset ``qs`` returns a particular list of values ``values``.
+    Asserts that a queryset ``qs`` matches a particular iterable of values
+    ``values``.
 
-    The comparison of the contents of ``qs`` and ``values`` is performed using
-    the function ``transform``; by default, this means that the ``repr()`` of
-    each value is compared. Any other callable can be used if ``repr()`` doesn't
-    provide a unique or helpful comparison.
+    If ``transform`` is supplied as a function, ``values`` is compared to a
+    list produced by applying ``transform`` to each member of ``qs``.
+
+    If ``transform`` is not supplied and ``values`` is a list of ``str``
+    elements, ``values`` is compared to a list produced by applying ``repr()``
+    to each member of ``qs``.
```

Comment:
I suggest we do it the other way around. If a `values` is a not list of `Model` instances then `repr` is applied to each member of `qs`. I also think we should deprecate this behaviour.

Reply:
We only want to use `repr` if `values` is a list of `str`s.